src/fasttext.cc

```diff
@@ -170,83 +178,70 @@ void FastText::test(std::istream& in, int32_t k) {
   std::cout << "Number of examples: " << nexamples << std::endl;
 }
 
-void FastText::predict(std::istream& in, int32_t k,
-                       std::vector<std::pair<real,std::string>>& predictions) const {
+bool FastText::predictNextLine(
+  std::istream &in,
+  int32_t k,
+  std::vector<std::pair<real, std::string>> &predictions
+) const {
+  if (in.peek(), in.eof()) {
+    return false;
+  }
+  predictions.clear();
   std::vector<int32_t> words, labels;
   dict_->getLine(in, words, labels, model_->rng);
   dict_->addNgrams(words, args_->wordNgrams);
-  if (words.empty()) return;
```

Comment:
Note that in the old code, early return here without `predictions.clear()` first might be buggy, since it could effectively inherit prediction from last time.